pkg/anchor/handler/proof/handler.go

```diff
@@ -42,24 +45,44 @@ type WitnessProofHandler struct {
 
 type witnessStore interface {
 	AddProof(vcID, witness string, p []byte) error
+	Get(vcID string) ([]*proofapi.WitnessProof, error)
 }
 
 type vcStore interface {
 	Get(id string) (*verifiable.Credential, error)
 }
 
+type vcStatusStore interface {
+	AddStatus(vcID string, status proofapi.VCStatus) error
+	GetStatus(vcID string) (proofapi.VCStatus, error)
+}
+
 type monitoringSvc interface {
 	Watch(vc *verifiable.Credential, endTime time.Time, domain string, created time.Time) error
 }
 
+type witnessPolicy interface {
+	Evaluate(witnesses []*proofapi.WitnessProof) (bool, error)
+}
+
 // HandleProof handles proof.
 func (h *WitnessProofHandler) HandleProof(witness *url.URL, anchorCredID string, endTime time.Time, proof []byte) error { //nolint:lll
 	logger.Debugf("received request anchorCredID[%s] from witness[%s], proof: %s",
 		anchorCredID, witness.String(), string(proof))
 
+	status, err := h.VCStatusStore.GetStatus(anchorCredID)
+	if err != nil {
+		return fmt.Errorf("failed to get status for anchor credential[%s]: %w", anchorCredID, err)
+	}
+
+	if status == proofapi.VCStatusCompleted {
+		// witness policy has been satisfied and witness proofs added to verifiable credential - nothing to do
```

Comment:
I wonder if we should still add the proof - even though the policy has been satisfied?

Reply:
When policy is satisfied, we store VC with proofs  into store and then we store anchor credential into CAS to get CID. At this point we should not add more proofs to VC because it will be different from the one stored in CAS.